src/vm/internal/otable.h

```diff
@@ -1,15 +1,51 @@
 #ifndef OTABLE_H
 #define OTABLE_H
 
-/*
- * otable.c
- */
-void init_otable(void);
-void enter_object_hash(struct object_t *);
-void remove_object_hash(struct object_t *);
-void remove_precompiled_hashes(char *);
-struct object_t *lookup_object_hash(const char *);
-int show_otable_status(struct outbuffer_t *, int);
-struct array_t *find_ch_n(const char *s);
+#include<cassert>
+#include<memory>
+#include<string>
+#include<unordered_map>
+#include<vector>
 
+#ifdef TESTING
+struct object_t
+{
+public:
+    std::string obname;
+
+    object_t(std::string const & s)
+    :obname(s) {}
+};
+#endif
+
+class ObjectTable /* final */
+{
+public:
+    using Key = std::string;
+    using Value = object_t*;
+    using Vector = std::vector<Value>;
```

Comment:
Key, Value aside, using Vector, M1, M2, Singleton here is just plain unnecessary , because it is mostly only used in this file , no other semantic or syntax value is added and making code unnecessarily hard to read. IMHO. 

Even for Key and Value it seems pretty unnecessary,  "using" statement should only be used if there are significant semantic (in an generic class maybe?) or textual value (very long types that are easy to get wrong) .  Here are neither of those cases.

Reply:
I think I agree with you here. M1,M2 probably isn't needed since it is only used once. Singleton also. I think Key,Value has been helpful though since I have changed the datatype and it only requires a 1 line change.

Reply:
Also vector was previously std::list. so again the Vector is convenient since it makes a datatype change a one liner as opposed to having to edit multiple points.

Reply:
Removed M1,M2.

Reply:
I was actually thinking of turning this into a template class "eventually" hence Key/Value. I think as a template the logic is clearer and it is also clearer in terms to what extent the dependencies in the code need to be hard coded. I plan however to do this later after I finish doing the living hash and probably the shared_string code.

Reply:
Ease of editing vs ease of reading are two different things, I would prefer latter almost always. As for future work, let's just focus on get this one in first, then worry about modifying it later.  

So, my suggestion is to ditch all using statement, plain and easy.

Reply:
Fine, I'm borderline okay with using Key Value and Vector here.